src/alchemlyb/convergence/convergence.py

```diff
@@ -389,3 +399,122 @@ def A_c(series_list, precision=0.01, tol=2):
+    num : int
+        The number of time points.
+    kwargs : dict
+        Keyword arguments to be passed to the estimator.
+
+    Returns
+    -------
+    :class:`pandas.DataFrame`
+        The DataFrame with estimate data. ::
+
+               FE             FE_Error
+            0  3.016442       0.052748
+            1  3.078106       0.037170
+            2  3.072561       0.030186
+            3  3.048325       0.026070
+            4  3.049769       0.023359
+            5  3.034078       0.021260
+            6  3.043274       0.019642
+            7  3.035460       0.018340
+            8  3.042032       0.017319
+            9  3.044149       0.016405
+
+
+    .. versionadded:: 2.4.0
+
+    """
+    logger.info("Start block averaging analysis.")
+    logger.info("Check data availability.")
+    if estimator not in (FEP_ESTIMATORS + TI_ESTIMATORS):
+        msg = f"Estimator {estimator} is not available in {FEP_ESTIMATORS + TI_ESTIMATORS}."
+        logger.error(msg)
+        raise ValueError(msg)
+    else:
+        # select estimator class by name
+        estimator_fit = estimators_dispatch[estimator](**kwargs).fit
+        logger.info(f"Use {estimator} estimator for convergence analysis.")
+
+    # Check that each df in the list has only one value of lambda
+    for i, df in enumerate(df_list):
+        lambda_values = list(set([x[1:] for x in df.index.to_numpy()]))
+        if len(lambda_values) > 1:
+            ind  = [j for j in range(len(lambda_values[0])) if len(list(set([x[j] for x in lambda_values]))) > 1][0]
+            raise ValueError(
+                "Provided DataFrame, df_list[{}] has more than one lambda value in df.index[{}]".format(i, ind)
+            )
+
+    if estimator in ["BAR"] and len(df_list) > 2:
+        raise ValueError(
+            "Restrict to two DataFrames, one with a fep-lambda value and one its forward adjacent state for a "
+            "meaningful result."
+        )
+        
+    # Choose length of comparison trajectory
+    lx_lambdas = [len(x) for x in df_list]
+    if len(set(lx_lambdas)) > 1:
+        lx = np.min( lx_lambdas)
+        warn(
+            "Not all trajectories for each lambda value are the same length, using minimum length for analysis: {}".format(
+                " ".join([f"len(df[{i}])={len(df_list[i])}" for i in range(len(df_list))])
+            ))
+    else:
+        lx = len(df_list[0])
+            
+    logger.info("Begin Moving Average Analysis")
+    average_list = []
+    average_error_list = []
+    for i in range(1, num):
+        logger.info("Moving Average Analysis: {:.2f}%".format(100 * i / num))
+        ind1, ind2 = lx // num * (i - 1), lx // num * i
+        sample = []
+        for data in df_list:
+            sample.append(data[ind1:ind2])
+        sample = concat(sample)
+        result = estimator_fit(sample)
+
+        average_list.append(result.delta_f_.iloc[0, -1])
+        if estimator.lower() == "bar":
```

Comment:
Might be good to move this part to the BAR estimator as we kind of do this in several places.

Reply:
Agreed!

Reply:
I'm fine with change that.

Reply:
I don't think this is something that should be changed in this PR. In the referenced issue comment Michael Shirts rejected a similar change because the standard deviations are correlated for BAR and a bootstrapping method is needed for accurate values, not a simple sqrt(sum(square())).

Reply:
I agree. This could be in a different PR.